### ISSUE TITLE
Automated testing

### ISSUE THREAD

**Issue by benwiley4000** [benwiley4000/react-responsive-audio-player#183]
* Run Nodejs-based tests using Travis or similar
* Run browser tests for all supported browsers (list to be determined based on whatever is currently usable or feasibly close) using BrowserStack

Depends on #35 being closed first.

**Comment by danielr18** [comment]
I would be happy to collaborate on this, but we would need to make some decisions on which testing framework to use (probably Jest), structure, main features to test, etc. If a checklist is created on how to start, I could help with the tasks and any other contributors as well.

**Comment by benwiley4000** [comment]
Awesome! I think Jest is a good tool to use. I could use some help brainstorming, but here's a starter checklist:

## Component snapshot tests
### `@cassette/player`
Each of these should be wrapped in context providers with mock values (the actual Context.Provider, not the smart wrapper):
- [ ] `MediaPlayerControls`
- [ ] All the control components
- Stuff that might be hard to test:
    - `VolumeControl` - can we mock different internal states?
### `@cassette/components
- [ ] `ProgressBarDisplay`
- [ ] Maybe `ProgressBar` (just to mock the `adjusting` state?)
- [ ] `MaybeMarquee`
- [ ] `VideoDisplay`
- Stuff that might be hard to test:
    - Potentially everything except for `ProgressBarDisplay`? These might be lower priority, especially since they're each so dependent on the DOM and on internal states.
## Unit tests for util functions
The various separated util functions could all have tests, and perhaps new util functions could be pulled out
## Context tests
It might require some refactoring and some mocks e.g. of the HTMLVideoElement, but it would be good to have API tests for the `playerContext` and `fullscreenContext`. I'm imagining that for the majority of functions available through the context, we have a test or tests that call the function then check afterward that:
1. The expected update occurred in the mock HTMLVideoElement
2. The expected change occurred in the context
Many of these updates will be asynchronous and some might even not be done on the next tick, so coding this could be tricky.

Mocking stuff from the browser will be an issue to resolve.

**Comment by benwiley4000** [comment]
I mentioned util tests, but thinking about it more I think the highest priority should be on context API testing, and component snapshots. I don't want to exert a bunch of effort writing tests for miscellaneous internal utilities that aren't even part of the API and might be changed or removed.

The last category of tests I forgot to mention is testing the props that get sent into the PlayerContextProvider, but this sounds super complex to me. If you have any great ideas I'd be glad to hear them, but I think it would be good to start with the two big categories I mentioned.

**Comment by benwiley4000** [comment]
Hm one more note, some of these tests could possibly be run in the browser (using Browser stack in CI) if it's not realistic to run them in Node. We have free Browserstack sponsorship for this project.

**Comment by benwiley4000** [comment]
We've already completed the first half of this, which is running our test suite in Travis.

The second half, running tests in our supported browsers with Browserstack, is a good idea but doesn't need to be finished for the beta, so I'll move this into the final 2.0 release target milestone.
